### PR TITLE
NPE while access background task count before it registered

### PR DESCRIPTION
**Pull request by anandjakhaniya**

### CODE REVIEW

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java [hunk 1]
```diff
@@ -586,13 +586,28 @@ public static Object retryOrThrowException(
     throw ex;
   }
 
+  public static void initializeBackgroundUtilsCount() {
+    try {
+      if (System.getProperties().get(ModelDBConstants.BACKGROUND_UTILS_COUNT) == null) {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]:
```suggestion
      if (System.getProperty(ModelDBConstants.BACKGROUND_UTILS_COUNT) == null) { 
```

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java [hunk 2]
```diff
@@ -586,13 +586,28 @@ public static Object retryOrThrowException(
...
+  }
+
   /**
    * If service want to call other verta service internally then should to registered those service
    * here with count
    */
   public static void registeredBackgroundUtilsCount() {
     int backgroundUtilsCount = 0;
-    if (System.getProperties().containsKey(ModelDBConstants.BACKGROUND_UTILS_COUNT)) {
+    if (System.getProperties().get(ModelDBConstants.BACKGROUND_UTILS_COUNT) != null) {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]:
```suggestion
    if (System.getProperty(ModelDBConstants.BACKGROUND_UTILS_COUNT) != null) {
```

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java [hunk 3]
```diff
@@ -602,7 +617,7 @@ public static void registeredBackgroundUtilsCount() {
 
   public static void unregisteredBackgroundUtilsCount() {
     int backgroundUtilsCount = 0;
-    if (System.getProperties().containsKey(ModelDBConstants.BACKGROUND_UTILS_COUNT)) {
+    if (System.getProperties().get(ModelDBConstants.BACKGROUND_UTILS_COUNT) != null) {
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]:
```suggestion
    if (System.getProperty(ModelDBConstants.BACKGROUND_UTILS_COUNT) != null) {
```

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java [hunk 4]
```diff
@@ -586,13 +586,28 @@ public static Object retryOrThrowException(
...
 
+  public static void initializeBackgroundUtilsCount() {
+    try {
+      if (System.getProperties().get(ModelDBConstants.BACKGROUND_UTILS_COUNT) == null) {
+        LOGGER.trace("Initialize runningBackgroundUtilsCount : {}", 0);
+        System.getProperties().put(ModelDBConstants.BACKGROUND_UTILS_COUNT, 0);
+      }
+      LOGGER.trace(
+          "Found runningBackgroundUtilsCount while initialization: {}",
+          System.getProperties().get(ModelDBConstants.BACKGROUND_UTILS_COUNT));
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]:
```suggestion
          System.getProperty(ModelDBConstants.BACKGROUND_UTILS_COUNT));
```

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java [hunk 5]
```diff
@@ -586,13 +586,28 @@ public static Object retryOrThrowException(
     throw ex;
   }
 
+  public static void initializeBackgroundUtilsCount() {
+    try {
+      if (System.getProperties().get(ModelDBConstants.BACKGROUND_UTILS_COUNT) == null) {
+        LOGGER.trace("Initialize runningBackgroundUtilsCount : {}", 0);
+        System.getProperties().put(ModelDBConstants.BACKGROUND_UTILS_COUNT, 0);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]:
```suggestion
        System.setProperty(ModelDBConstants.BACKGROUND_UTILS_COUNT, "0");
```

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java [hunk 6]
```diff
@@ -586,13 +586,28 @@ public static Object retryOrThrowException(
...
+      }
+      LOGGER.trace(
+          "Found runningBackgroundUtilsCount while initialization: {}",
+          System.getProperties().get(ModelDBConstants.BACKGROUND_UTILS_COUNT));
+    } catch (NullPointerException ex) {
+      LOGGER.trace("NullPointerException while initialize runningBackgroundUtilsCount");
+      System.getProperties().put(ModelDBConstants.BACKGROUND_UTILS_COUNT, 0);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]:
```suggestion
      System.setProperty(ModelDBConstants.BACKGROUND_UTILS_COUNT, "0");
```

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java [hunk 7]
```diff
@@ -611,10 +626,16 @@ public static void unregisteredBackgroundUtilsCount() {
   }
 
   public static int getRegisteredBackgroundUtilsCount() {
-    int backgroundUtilsCount =
-        (int) System.getProperties().get(ModelDBConstants.BACKGROUND_UTILS_COUNT);
-    LOGGER.trace("get runningBackgroundUtilsCount : {}", backgroundUtilsCount);
-    return backgroundUtilsCount;
+    try {
+      int backgroundUtilsCount =
+          (int) System.getProperties().get(ModelDBConstants.BACKGROUND_UTILS_COUNT);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]:
```suggestion
          (int) System.getProperty(ModelDBConstants.BACKGROUND_UTILS_COUNT);
```

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java [hunk 8]
```diff
@@ -611,10 +626,16 @@ public static void unregisteredBackgroundUtilsCount() {
...
+      LOGGER.trace("get runningBackgroundUtilsCount : {}", backgroundUtilsCount);
+      return backgroundUtilsCount;
+    } catch (NullPointerException ex) {
+      LOGGER.trace("NullPointerException while get runningBackgroundUtilsCount");
+      System.getProperties().put(ModelDBConstants.BACKGROUND_UTILS_COUNT, 0);
```
Comment (ravishetye) [inline backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]:
```suggestion
      System.setProperty(ModelDBConstants.BACKGROUND_UTILS_COUNT, "0");
```

**Review by ravishetye — CHANGES_REQUESTED** [review]
this can have improvements. Please check the changes.
To resolve the NPE I will update add a try-catch in a different PR.

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java (hunk 1 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java (hunk 3 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java (hunk 4 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java (hunk 5 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java (hunk 6 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java (hunk 7 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]: Fixed

File: backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java (hunk 8 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java]: Fixed

**Review by ravishetye — APPROVED** [review]

**Pull request merged**